tools/hardware/device.py

```diff
@@ -135,12 +135,9 @@ def get_interrupt_affinities(self) -> List[int]:
 
     def visit(self, visitor: Any):
         ''' Visit this node and all its children '''
-        ret = [visitor(self)]
-        if ret[0] is None:
-            ret = []
+        visitor(self)
         for child in self.children.values():
-            ret += child.visit(visitor)
-        return ret
+            child.visit(visitor)
```

Comment:
This changes the semantics of `visit`, we're not returning the list of visited nodes any more. Why is this safe? Do we know all callers of this function?

Reply:
I don't understand the commit message on this one: the previous implementation was perfectly sane in constructing the list of return values of the visited elements, leaving out `None` values. I'm not sure why that was done, but there is nothing specifically wrong with it.

What was the linter complaint for it?

Reply:
I think I got this pattern wrong in the end, when trying to define the return type properly. The return type of the custom `visitor()` function is simply an `object` and `visit()` returns a list of objects. The linter should accept this - will check again if there was some issues here.